filebeat/fileset/factory.go

```diff
@@ -77,15 +77,9 @@ func NewFactory(
 
 // Create creates a module based on a config
 func (f *Factory) Create(p beat.PipelineConnector, c *common.Config) (cfgfile.Runner, error) {
-	// Start a registry of one module:
-	m, err := NewModuleRegistry([]*common.Config{c}, f.beatInfo, false)
-	if err != nil {
-		return nil, err
-	}
-
-	pConfigs, err := m.GetInputConfigs()
+	m, pConfigs, err := f.createRegistry(c)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("create module registry for filesets: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("couldn't create module registry for filesets: %w", err)
```

Reply:
+1 on changing the wording that something did not work out

Disclaimer: I'm no native speaker

I'd prefer if our logs do not use `didn't`, `couldn't` or similar. Instead put some emphasis  on the `not` by using `could not`, `can not`, `must not`.

Reply:
👍🏼 edited to use the full form of negation

Reply:
Ok, adding `could not`.

I have been changing my opinion on these kind of messages over time. Lately I am wondering if, when wrapping, it is not enough to mention what was being done. The wrapped error `err` already has the root cause, and the caller knows how it affects upper layers or the user, the context provided in places like here give an orientation of what was being done.

Thinking in a made up example, these are the kind of messages logged when one or the other strategy are used:
* `Failed to read configuration: could not read configuration from file /somefile: could not open file: permission denied`.
* `Failed to read configuration: read configuration from file /somefile: open file: permission denied`.

Both messages provide the same information: what failed, what was it trying to do and what was the root cause of the issue. In the first version the `could not`s don't provide additional information, and when many errors are chained they can look too redundant. Second option looks enough, more compact and easier to read.

In the code, the message looks more complete when mentioning that something didn't went as expected, but this information is actually redundant with the fact that an error is being wrapped and returned.

Reply:
Well, part of the problem is we do not capture stack traces (or locations) on errors + generic error types normally do not carry additional 'context' (and we do not define our own error types when we should). Plus there is no generic support to add diagnostic context to add extra information to an error value (I'm using [sderr](https://pkg.go.dev/github.com/urso/sderr/) like `sderr.With("file", ...).Wrap(err, "read configuration")` if I need this). Aaannnd we do not spend too much time thinking how the final log message might look like :)

If you look at the error from the stdlib, it includes the operation and the cause like `open file: permission denied`. In order to create user-readable errors one might use an encoding like `<operation>: <optional context>: <optional message>: <cause>`, where `<cause>` should follow a similar pattern. 

Plus our logs are sometimes redundant to our error message. Following the aforementioned schema + reduce redundancy in the log (we do not need to mention that "reading configuration" failed), the log message would become:

`Error: read configuration: filename /somefile: open file: permission denied`.

Beside the `Error` and the `permission denied` there is no mention of `failing`, `can not`, or similar. The message more or less resembles that call stack (at the cost of having "readable" english).

But our logs/errors normally follow the schema `Error doing X: failed to <message similar to X>: could not <message>: <cause>`. Although it does not read nice, we start each sentence with "failed" and might want to keep that.

Some related readings:
- https://commandcenter.blogspot.com/2017/12/error-handling-in-upspin.html
- https://middlemost.com/failure-is-your-domain/
- [go-txfile error doc](https://github.com/elastic/go-txfile/blob/master/txerr/doc.go)